Makefile

```diff
@@ -796,7 +813,19 @@ $(LIBDIR)/libsameboy.a: $(LIBDIR)/libsameboy.o
 	-@$(MKDIR) -p $(dir $@)
 	-@rm -f $@
 	ar -crs $@ $^
-	
+
+$(LIBDIR)/libsameboy.so: $(LIBDIR)/libsameboy.o
```

Comment:
The library should be versioned with appropriate symlinks.

Reply:
Also different platforms have unique behavior with shared libraries, the jgemu build systems have tried to handle this.

https://gitlab.com/jgemu/bsnes/-/blob/5376dc9e2154def98a3d1c4b867ccb7c632baaf7/mk/jg.mk#L33-65

Reply:
> The library should be versioned with appropriate symlinks.

I'm not familiar with this; are symlinks by themselves enough, or is something more needed? I see GNU ld has the ability to version symbols in a library, but that's probably overkill here?

Reply:
And should the API version track the library version? That'd make sense to me, but in practice I've seen both mismatch often, it seems.

Reply:
A link explaining best practices for C library versioning would help, if you have one.

Reply:
That's the only unresolved thing.

Reply:
Take a look at other libraries on the system, like for example libpkgconf.
```
$ ls -l libpkgconf.so*
lrwxrwxrwx 1 root root    19 Oct  6 06:33 libpkgconf.so -> libpkgconf.so.5.0.0
lrwxrwxrwx 1 root root    19 Oct  6 06:33 libpkgconf.so.5 -> libpkgconf.so.5.0.0
-rwxr-xr-x 1 root root 72848 Oct  6 06:33 libpkgconf.so.5.0.0
```
Where the real file has the full version and there are symlinks for the non-versioned library and only the major version. Note that Darwin has a different scheme for the order of the extension and version numbers. And for modules (As opposed to shared libraries) that are loaded at runtime like with JGEMU or Libretro they only have the extension and no version.

[semver.org](https://semver.org/) explains some of this.

And yes you should also use `-soname` to version the library, so something like `-Wl,-soname,$(LIB_MAJOR)` while Darwin would use `-Wl,-install_name,$(LIBDIR)/$(LIB_MAJOR)`.

As for the API version I would think its fine to follow the program version here, but @LIJI32 would ultimately decide what is best here.

Reply:
I felt like produced versioned shared libraries properly on various systems was enough of a pain to justify the introduction of a dependency: libtool. My latest revision makes use of it; hopefully it'll unlock building libraries for Windows as well.

I've removed some LTO-related hack which I assume libtool will take care of. This will need more testing to ensure I haven't broken anything on all platforms.